rust/client/src/lib.rs

```diff
@@ -324,6 +297,127 @@ impl BpxClient {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct BpxClientBuilder {
+    base_url: Option<String>,
+    ws_url: Option<String>,
+    secret: Option<String>,
+    headers: Option<BpxHeaders>,
+}
+
+impl BpxClientBuilder {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Sets the base URL for the Backpack Exchange API.
+    /// If not set, defaults to `BACKPACK_API_BASE_URL`.
+    ///
+    /// # Arguments
+    /// * `base_url` - The base URL
+    ///
+    /// # Returns
+    /// * `Self` - The updated builder instance
+    pub fn base_url(mut self, base_url: impl ToString) -> Self {
+        self.base_url = Some(base_url.to_string());
+        self
+    }
+
+    /// Sets the WebSocket URL for the Backpack Exchange API.
+    /// If not set, defaults to `BACKPACK_WS_URL`.
+    ///
+    /// # Arguments
+    /// * `ws_url` - The WebSocket URL
+    ///
+    /// # Returns
+    /// * `Self` - The updated builder instance
+    pub fn ws_url(mut self, ws_url: impl ToString) -> Self {
+        self.ws_url = Some(ws_url.to_string());
+        self
+    }
+
+    /// Sets the API secret for signing requests.
+    /// If not set, the client will be unauthenticated.
+    ///
+    /// # Arguments
+    /// * `secret` - The API secret
+    ///
+    /// # Returns
+    /// * `Self` - The updated builder instance
+    pub fn secret(mut self, secret: impl ToString) -> Self {
+        self.secret = Some(secret.to_string());
+        self
+    }
+
+    /// Sets custom HTTP headers for the client.
+    /// If not set, no additional headers will be included.
+    ///
+    /// # Arguments
+    /// * `headers` - The custom HTTP headers
+    ///
+    /// # Returns
+    /// * `Self` - The updated builder instance
+    pub fn headers(mut self, headers: BpxHeaders) -> Self {
+        self.headers = Some(headers);
+        self
+    }
+
+    /// Builds the `BpxClient` instance with the configured parameters.
+    ///
+    /// # Returns
+    /// * `Result<BpxClient>` - The constructed client or an error if building fails
+    pub fn build(self) -> Result<BpxClient> {
+        let base_url = self
+            .base_url
+            .map(Cow::from)
+            .unwrap_or_else(|| Cow::Borrowed(BACKPACK_API_BASE_URL));
+
+        let base_url = Url::parse(&base_url).map_err(|e| Error::UrlParseError(e.to_string().into()))?;
+
+        let ws_url = self
+            .ws_url
+            .map(Cow::from)
+            .unwrap_or_else(|| Cow::Borrowed(BACKPACK_WS_URL));
+
+        let ws_url = Url::parse(&ws_url).map_err(|e| Error::UrlParseError(e.to_string().into()))?;
```

Comment:
No need for `Cow` here:

```suggestion
        let base_url = self.base_url.as_deref().unwrap_or(BACKPACK_API_BASE_URL);
        let base_url = Url::parse(base_url).map_err(|e| Error::UrlParseError(e.to_string().into()))?;

        let ws_url = self.ws_url.as_deref().unwrap_or(BACKPACK_WS_URL);
        let ws_url = Url::parse(ws_url).map_err(|e| Error::UrlParseError(e.to_string().into()))?;
```